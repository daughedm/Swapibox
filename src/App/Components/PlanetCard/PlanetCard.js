import React from 'react';
import '../PlanetCard/PlanetCard.css';
import heart from '../../../Assets/Images/like.svg';
import PropTypes from 'prop-types';

const PlanetCard = ({ Data, image, addFavorite }) => {
  return (
    <div className="card-planet" >
      <div  
        className={image} 
        alt="planet">
        <div  
          className="favorite"
          onClick={() => addFavorite(Data)}>
          <img  
            className="fav-icon" 
            src={heart} 
            alt="favorite" />
        </div>
      </div>
      <h3 className="card-header">{Data.name}</h3>
      <h5 className="sub-categories top">Climate: 
        <span className="normal"> {Data.climate}</span>
      </h5>
      <h5 className="sub-categories">Terrain: 
        <span className="normal"> {Data.terrain}</span>
      </h5>
      <h5 className="sub-categories">Population: 
        <span className="normal"> {Data.population}</span>
      </h5>
      <h5 className="sub-categories">Residents: 
        <span className="normal"> {Data.residents}</span>
      </h5>
    </div>
  );
};

PlanetCard.propTypes = {
  Data: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.array,
    PropTypes.object,
  ]),
  image: PropTypes.string,
  addFavorite: PropTypes.func
};


export default PlanetCard;