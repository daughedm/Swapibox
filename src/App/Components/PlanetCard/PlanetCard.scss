.card {
  background-color: white;
  width: 250px;
  height: 370px;
  padding: 15px;
  margin: 15px;
  overflow: auto;
}

.people0 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/lukeskywalker.jpg);
  background-size: cover;
}

.people1 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/C3PO.jpg);
  background-size: cover;
}

.people2 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/R2D2.jpg);
  background-size: cover;
}

.people3 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/DarthVader.jpg);
  background-size: cover;
}

.people4 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/leia.jpg);
  background-size: cover;
}

.people5 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/owenlars.jpg);
  background-size: cover;
}

.people6 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/beru.jpg);
  background-size: cover;
}

.people7 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/R5D4.jpg);
  background-size: cover;
}

.people8 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/biggs.jpg);
  background-size: cover;
}

.people9 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/Obiwan.jpg);
  background-size: cover;
}

.planets0 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/alderaan.jpg);
  background-size: cover;
}

.planets1 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/Yavin.jpg);
  background-size: cover;
}

.planets2 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/Hoth.jpg);
  background-size: cover;
}

.planets3 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/Dagobah.jpg);
  background-size: cover;
}

.planets4 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/Bespin.jpg);
  background-size: cover;
}

.planets5 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/endor.jpg);
  background-size: cover;
}

.planets6 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/naboo.jpg);
  background-size: cover;
}

.planets7 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/Coruscant.jpg);
  background-size: cover;
}

.planets8 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/kamino.jpg);
  background-size: cover;
}

.planets9 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/geonosis.jpg);
  background-size: cover;
}

.vehicle0 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/sandcrawler.jpg);
  background-size: cover;
}

.vehicle1 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/skyhopper.jpg);
  background-size: cover;
}

.vehicle2 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/landspeeder.jpg);
  background-size: cover;
}

.vehicle3 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/tie-fighter.jpg);
  background-size: cover;
}

.vehicle4 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/Snowspeeder.jpg);
  background-size: cover;
}

.vehicle5 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/tie-bomber.jpg);
  background-size: cover;
}

.vehicle6 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/ATAT.jpg);
  background-size: cover;
}

.vehicle7 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/ATST.jpg);
  background-size: cover;
}

.vehicle8 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/twinpod.jpg);
  background-size: cover;
}

.vehicle9 {
  width: 100%;
  height: 140px;
  background-image: url(../../../Assets/Images/sailbarge.jpg);
  background-size: cover;
}