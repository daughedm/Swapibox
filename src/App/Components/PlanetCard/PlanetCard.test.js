import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import PlanetCard from './PlanetCard';

describe('PlanetCard', () => {
  let wrapper;

  it('should match snapshot', () => {
    const planet = {
      "name": "Alderaan",
      "terrain": "grasslands, mountains",
      "climate": "temperate",
      "population": "2000000000",
      "residents": 'Bob'
    };
    const addFavorite = jest.fn();
    wrapper = shallow(<PlanetCard item={planet}
      Data="planet"
      addFavorite={addFavorite} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should call addFavorite on click with the correct props', () => {
    let mockProps = {
      Data: {
        "name": "Alderaan",
        "terrain": "grasslands, mountains",
        "climate": "temperate",
        "population": "2000000000",
        "residents": 'Bob'
      },
      addFavorite: jest.fn()
    };

    const wrapper = shallow(<PlanetCard {...mockProps} />);

    wrapper.find('.favorite').simulate('click');

    expect(mockProps.addFavorite).toHaveBeenCalledTimes(1);
    expect(mockProps.addFavorite).toHaveBeenCalledWith(mockProps.Data);

  });
});