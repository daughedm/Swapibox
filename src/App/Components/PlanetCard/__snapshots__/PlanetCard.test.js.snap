// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanetCard should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlanetCard
    Data="planet"
    addFavorite={[Function]}
    item={
        Object {
            "climate": "temperate",
            "name": "Alderaan",
            "population": "2000000000",
            "residents": "Bob",
            "terrain": "grasslands, mountains",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          alt="planet"
          className={undefined}
>
          <div
                    className="favorite"
                    onClick={[Function]}
          >
                    <img
                              alt="favorite"
                              className="fav-icon"
                              src="like.svg"
                    />
          </div>
</div>,
        <h3
          className="card-header"
/>,
        <h5
          className="sub-categories top"
>
          Climate:
          <span
                    className="normal"
          >
                     
          </span>
</h5>,
        <h5
          className="sub-categories"
>
          Terrain:
          <span
                    className="normal"
          >
                     
          </span>
</h5>,
        <h5
          className="sub-categories"
>
          Population:
          <span
                    className="normal"
          >
                     
          </span>
</h5>,
        <h5
          className="sub-categories"
>
          Residents:
          <span
                    className="normal"
          >
                     
          </span>
</h5>,
      ],
      "className": "card-planet",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "planet",
          "children": <div
            className="favorite"
            onClick={[Function]}
>
            <img
                        alt="favorite"
                        className="fav-icon"
                        src="like.svg"
            />
</div>,
          "className": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="favorite"
              className="fav-icon"
              src="like.svg"
/>,
            "className": "favorite",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "favorite",
              "className": "fav-icon",
              "src": "like.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "card-header",
        },
        "ref": null,
        "rendered": null,
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Climate:",
            <span
              className="normal"
>
               
</span>,
          ],
          "className": "sub-categories top",
        },
        "ref": null,
        "rendered": Array [
          "Climate:",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                undefined,
              ],
              "className": "normal",
            },
            "ref": null,
            "rendered": Array [
              " ",
              undefined,
            ],
            "type": "span",
          },
        ],
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Terrain:",
            <span
              className="normal"
>
               
</span>,
          ],
          "className": "sub-categories",
        },
        "ref": null,
        "rendered": Array [
          "Terrain:",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                undefined,
              ],
              "className": "normal",
            },
            "ref": null,
            "rendered": Array [
              " ",
              undefined,
            ],
            "type": "span",
          },
        ],
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Population:",
            <span
              className="normal"
>
               
</span>,
          ],
          "className": "sub-categories",
        },
        "ref": null,
        "rendered": Array [
          "Population:",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                undefined,
              ],
              "className": "normal",
            },
            "ref": null,
            "rendered": Array [
              " ",
              undefined,
            ],
            "type": "span",
          },
        ],
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Residents:",
            <span
              className="normal"
>
               
</span>,
          ],
          "className": "sub-categories",
        },
        "ref": null,
        "rendered": Array [
          "Residents:",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                undefined,
              ],
              "className": "normal",
            },
            "ref": null,
            "rendered": Array [
              " ",
              undefined,
            ],
            "type": "span",
          },
        ],
        "type": "h5",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            alt="planet"
            className={undefined}
>
            <div
                        className="favorite"
                        onClick={[Function]}
            >
                        <img
                                    alt="favorite"
                                    className="fav-icon"
                                    src="like.svg"
                        />
            </div>
</div>,
          <h3
            className="card-header"
/>,
          <h5
            className="sub-categories top"
>
            Climate:
            <span
                        className="normal"
            >
                         
            </span>
</h5>,
          <h5
            className="sub-categories"
>
            Terrain:
            <span
                        className="normal"
            >
                         
            </span>
</h5>,
          <h5
            className="sub-categories"
>
            Population:
            <span
                        className="normal"
            >
                         
            </span>
</h5>,
          <h5
            className="sub-categories"
>
            Residents:
            <span
                        className="normal"
            >
                         
            </span>
</h5>,
        ],
        "className": "card-planet",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "planet",
            "children": <div
              className="favorite"
              onClick={[Function]}
>
              <img
                            alt="favorite"
                            className="fav-icon"
                            src="like.svg"
              />
</div>,
            "className": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="favorite"
                className="fav-icon"
                src="like.svg"
/>,
              "className": "favorite",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "favorite",
                "className": "fav-icon",
                "src": "like.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "card-header",
          },
          "ref": null,
          "rendered": null,
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Climate:",
              <span
                className="normal"
>
                 
</span>,
            ],
            "className": "sub-categories top",
          },
          "ref": null,
          "rendered": Array [
            "Climate:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  undefined,
                ],
                "className": "normal",
              },
              "ref": null,
              "rendered": Array [
                " ",
                undefined,
              ],
              "type": "span",
            },
          ],
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Terrain:",
              <span
                className="normal"
>
                 
</span>,
            ],
            "className": "sub-categories",
          },
          "ref": null,
          "rendered": Array [
            "Terrain:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  undefined,
                ],
                "className": "normal",
              },
              "ref": null,
              "rendered": Array [
                " ",
                undefined,
              ],
              "type": "span",
            },
          ],
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Population:",
              <span
                className="normal"
>
                 
</span>,
            ],
            "className": "sub-categories",
          },
          "ref": null,
          "rendered": Array [
            "Population:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  undefined,
                ],
                "className": "normal",
              },
              "ref": null,
              "rendered": Array [
                " ",
                undefined,
              ],
              "type": "span",
            },
          ],
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Residents:",
              <span
                className="normal"
>
                 
</span>,
            ],
            "className": "sub-categories",
          },
          "ref": null,
          "rendered": Array [
            "Residents:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  undefined,
                ],
                "className": "normal",
              },
              "ref": null,
              "rendered": Array [
                " ",
                undefined,
              ],
              "type": "span",
            },
          ],
          "type": "h5",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
